Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -3,365 +3,2701 @@
+
+In some cases, you might find data in `userInfo` of the `NSUserActivity` object. In the previous case there was no data being transferred but it might be the case for other scenarios. To see this, be sure to hook the `userInfo` property or access it directly from the `continueUserActivity` object in your hook (e.g. by adding a line like this `log("userInfo:" + ObjC.Object(args[3]).userInfo().toString());`).
+
+**Final Notes about Universal Links and Handoff**
+
+What we have seen above is also [the way Handoff works](https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/Handoff/HandoffFundamentals/HandoffFundamentals.html#//apple_ref/doc/uid/TP40014338 "Handoff Fundamentals: About Handoff") and how you could also see the received data. Actually, this is very similar to the ["Web Browser–to–Native App Handoff"](https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/Handoff/AdoptingHandoff/AdoptingHandoff.html#//apple_ref/doc/uid/TP40014338-CH2-SW10 "Adopting Handoff: Web Browser–to–Native App") scenario:
+
+> If the user is using a web browser on the originating device, and the receiving device is an iOS device with a native app that claims the domain portion of the `webpageURL` property, then iOS launches the native app and sends it an `NSUserActivity` object with an `activityType` value of `NSUserActivityTypeBrowsingWeb`. The `webpageURL` property contains the URL the user was visiting, while the `userInfo` dictionary is empty.
+
+Like universal links, the Handoff Activity Continuation is defined in the `com.apple.developer.associated-domains` entitlement and in the `apple-app-site-association` file from the server, both using the keyword `"activitycontinuation":`. See "Retrieving the App Site Association File" above for an example.
+
+
+#### UIActivity Sharing
+
+##### Overview
+
+Starting on iOS 6 it is possible for third-party apps to share information via specific mechanisms [like AirDrop, for example](https://developer.apple.com/library/archive/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/Inter-AppCommunication/Inter-AppCommunication.html#//apple_ref/doc/uid/TP40007072-CH6-SW3 "Supporting AirDrop"). From a user perspective, this feature is the well-known system-wide *share activity sheet* that appears after clicking on the "Share" button.
+
+![Share Activity Sheet](Images/Chapters/0x06h/share_activity_sheet.png)
+
+The available sharing mechanisms (aka. Activity Types) include:
+
+- airDrop
+- assignToContact
+- copyToPasteboard
+- mail
+- message
+- postToFacebook
+- postToTwitter
+
```

Comment:
I should then change the overview a little bit as I see that it can be misleading with those items.

The user clicks on "Share a picture" from your app. iOS by default will present these built-in mechanisms plus all other apps that support a picture. There's nothing specific that your app can do about those built-in mechanisms apart from excluding them if they are not appropriate (e.g. you don't want your users to post to Twitter explicitly.

The static and dynamic analysis "Sending/Receiving Files" sections are about your app sending and receiving data via UIActivityViewController. 

On the "Sending" side, you can set which data: maybe a photo or a set of data to be shared (weblink, photo and text) and maybe exclude some sharing mechanisms (from the ones mentioned above). The user will select one from the presented options but actually is the receiving app the one that has to check the transmitted data, e.g. Facebook or Twitter. 

On the "Receiving" side, if we are the other app and we are receiving data, then the "Receiving Files" sections already explain what to check exactly. 

Maybe I should rename the sections "Sending/Receiving Files" to "Sending/Receiving Items" to avoid confusion.

Reply:
Makes sense to indeed cut it into better pieces and then still describe it. Can you make a separate issue out of that which you can fix in another pr? otherwise this will stale too much :(.